### PR TITLE
Add communications identity types

### PR DESCRIPTION
**Pull request by Hunterjet**
Add communications identity types. Update some files referencing the endpointType enum.

### CODE REVIEW

**Review by mkhribech — APPROVED** [review]

File: api-reference/beta/resources/communicationsidentityset.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+| encrypted | [communicationsEncryptedIdentity](communicationsEncryptedIdentity.md) | The encrypted user associated with this action. |
+| azureCommunicationServicesUser | [azureCommunicationServicesUserIdentity](azureCommunicationServicesUserIdentity.md) | The Azure Communication Services user associated with this action. |
+| applicationInstance | [communicationsApplicationInstanceIdentity](communicationsApplicationInstanceIdentity.md) | The application instance associated with this action. |
+| onPremises | [communicationsUserIdentity](communicationsUserIdentity.md) | The Skype for Business On-Premises user associated with this action. |
+| assertedIdentity | [communicationsUserIdentity](communicationsUserIdentity.md) or [communicationsPhoneIdentity](communicationsPhoneIdentity.md) | An **identity** the participant would like to present itself as to the other participants in the call. |
+| endpointType | [endpointType](endpointType.md) | Type of endpoint the participant is using. Possible values are: `default`, `voicemail`, `skypeForBusiness`, `skypeForBusinessVoipPhone`. |
```
Comment (isvargasmsft) [inline api-reference/beta/resources/communicationsidentityset.md:33]: We always add `unknownFutureValue` for evolvable enums. Guidance here: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211749/Evolvable-enumerations

```suggestion
| endpointType | [endpointType](endpointType.md) | Type of endpoint the participant is using. Possible values are: `default`, `voicemail`, `skypeForBusiness`, `skypeForBusinessVoipPhone` and `unknownFutureValue`. |
```

File: api-reference/beta/resources/endpointtype.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+--- 
+title: "endpointType resource type"
+description: "Represents the type of endpoint being used by a participant of a communication."
+author: "mkhribech"
+ms.localizationpriority: medium
+ms.prod: "cloud-communications"
+doc_type: resourcePageType
+---
+
+# endpointType resource type
```
Comment (isvargasmsft) [inline api-reference/beta/resources/endpointtype.md:10]: I don't see any new (or pre-existing) resource types with this name in the CSDL. Only the preexisting enum. Could you please clarify this topic?

File: api-reference/beta/resources/invitationparticipantinfo.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ Represents an entity that is being invited to a group call.
 
 | Property                           | Type                          | Description                                                                          |
 | :--------------------------------- | :---------------------------- | :----------------------------------------------------------------------------------- |
-| endpointType                       | String                        | The type of the endpoint. Possible values are: `default`, `voicemail`. |
+| endpointType                       | endpointType                        | The type of endpoint. Possible values are: `default`, `voicemail`, `skypeForBusiness`, `skypeForBusinessVoipPhone`. |
```
Comment (isvargasmsft) [inline api-reference/beta/resources/invitationparticipantinfo.md:22]:
```suggestion
| endpointType                       | endpointType                        | The type of endpoint. Possible values are: `default`, `voicemail`, `skypeForBusiness`, `skypeForBusinessVoipPhone` and `unknownFutureValue`. |
```

File: api-reference/beta/resources/invitationparticipantinfo.md [hunk 4]
```diff
@@ -38,7 +38,7 @@ The following is a JSON representation of the resource.
 }-->
 ```json
 {
-  "endpointType": "String",
+  "endpointType": "default | voicemail | skypeForBusiness | skypeForBusinessVoipPhone",
```
Comment (isvargasmsft) [inline api-reference/beta/resources/invitationparticipantinfo.md:41]: In JSON representations, please use String for enum types. 
```suggestion
  "endpointType": "String",
```

File: api-reference/beta/resources/participantinfo.md [hunk 5]
```diff
@@ -48,7 +48,7 @@ The following is a JSON representation of the resource.
 {
   "countryCode": "String",
   "identity": { "@odata.type": "#microsoft.graph.identitySet" },
-  "endpointType": "default | skypeForBusiness | skypeForBusinessVoipPhone",
+  "endpointType": "default | voicemail | skypeForBusiness | skypeForBusinessVoipPhone",
```
Comment (isvargasmsft) [inline api-reference/beta/resources/participantinfo.md:51]:
```suggestion
  "endpointType": "String",
```

File: api-reference/beta/resources/communicationsidentityset.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  "application": {"@odata.type": "microsoft.graph.communicationsApplicationIdentity"},
+  "user": {"@odata.type": "microsoft.graph.communicationsUserIdentity"},
+  "phone": {"@odata.type": "microsoft.graph.communicationsPhoneIdentity"},
+  "guest": {"@odata.type": "microsoft.graph.communicationsGuestIdentity"},
+  "encrypted": {"@odata.type": "microsoft.graph.communicationsEncryptedIdentity"},
+  "azureCommunicationServicesUser": {"@odata.type": "microsoft.graph.azureCommunicationServicesUserIdentity"},
+  "applicationInstance": {"@odata.type": "microsoft.graph.communicationsApplicationInstanceIdentity"},
+  "onPremises": {"@odata.type": "microsoft.graph.communicationsUserIdentity"},
+  "assertedIdentity": {"@odata.type": "microsoft.graph.identity"},
+  "endpointType": "default | voicemail | skypeForBusiness | skypeForBusinessVoipPhone"
```
Comment (isvargasmsft) [inline api-reference/beta/resources/communicationsidentityset.md:66]:
```suggestion
  "endpointType": "String"
```

File: api-reference/beta/resources/communicationsapplicationinstanceidentity.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+doc_type: resourcePageType
+---
+
+# communicationsApplicationInstanceIdentity resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the identity of an instance of an application. An application instance is a resource account created by **New-CsOnlineApplicationInstance**. For example, you might have multiple Call Queue application instances that are all backed by the same Azure Active Directory application. These application instances are represented in Azure Active Directory by disabled user accounts.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/communicationsapplicationinstanceidentity.md:16]: Do we have a reference link we can provide to give more context on what is **New-CsOnlineApplicationInstance**?

File: api-reference/beta/resources/communicationsapplicationidentity.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+doc_type: resourcePageType
+---
+
+# communicationsApplicationIdentity resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the identity of an application.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/communicationsapplicationidentity.md:16]: Would you please add a bit more of context to this description?

File: api-reference/beta/resources/communicationsuseridentity.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+doc_type: resourcePageType
+---
+
+# communicationsUserIdentity resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the identity of a user.
```
Comment (isvargasmsft) [inline api-reference/beta/resources/communicationsuseridentity.md:16]: Would you please provide a bit more of context to this description as well?

**Review by isvargasmsft — CHANGES_REQUESTED** [review]
Hello @Hunterjet, 

Thank you very much for your work in this PR. Very complete in general. 

1. Please take a look at a few suggestions I left for consistency. 
2. Please clarify why we are adding a resource topic for endpointType. 
3. Please provide a bit more of context to 2 descriptions. 

Kind regards, 

Isaac

File: api-reference/beta/resources/communicationsidentityset.md (hunk 1 above)
Comment (Hunterjet) [reply api-reference/beta/resources/communicationsidentityset.md]: Added.

File: api-reference/beta/resources/endpointtype.md (hunk 2 above)
Comment (Hunterjet) [reply api-reference/beta/resources/endpointtype.md]: My bad, this was from an earlier iteration. Removed the file.

File: api-reference/beta/resources/invitationparticipantinfo.md (hunk 3 above)
Comment (Hunterjet) [reply api-reference/beta/resources/invitationparticipantinfo.md]: Added.

File: api-reference/beta/resources/participantinfo.md [hunk 10]
```diff
@@ -20,7 +20,7 @@ Contains additional properties about the participant identity
...
 | :--------------- | :------------------------------ | :-----------------------------------------------------------------------------------------------------------------------------------------------------------                                                     |
 | countryCode      | String                          | The ISO 3166-1 Alpha-2 country code of the participant's best estimated physical location at the start of the call. Read-only.                                                                                   |
-| endpointType     | String                          | The type of endpoint the participant is using. Possible values are: `default`, `skypeForBusiness`, or `skypeForBusinessVoipPhone`. Read-only.                                                                    |
+| endpointType     | endpointType                          | The type of endpoint the participant is using. Possible values are: `default`, `voicemail`, `skypeForBusiness`, `skypeForBusinessVoipPhone`. Read-only.                                                                    |
```
[the remark this answers is not in the document]
Comment (Hunterjet) [reply api-reference/beta/resources/participantinfo.md]: Added.

File: api-reference/beta/resources/communicationsidentityset.md (hunk 6 above)
Comment (Hunterjet) [reply api-reference/beta/resources/communicationsidentityset.md]: Fixed.

File: api-reference/beta/resources/communicationsapplicationinstanceidentity.md (hunk 7 above)
Comment (Hunterjet) [reply api-reference/beta/resources/communicationsapplicationinstanceidentity.md]: Added link to the docs page for this cmdlet.

File: api-reference/beta/resources/communicationsapplicationidentity.md (hunk 8 above)
Comment (Hunterjet) [reply api-reference/beta/resources/communicationsapplicationidentity.md]: Added.

File: api-reference/beta/resources/communicationsuseridentity.md (hunk 9 above)
Comment (Hunterjet) [reply api-reference/beta/resources/communicationsuseridentity.md]: Added.

**Review by isvargasmsft — APPROVED** [review]

**Comment by isvargasmsft** [comment]
@FaithOmbongi FYI I just reviewed this one :)

**Pull request merged**